app/lib/features/events/pages/event_details_page.dart

```diff
@@ -62,52 +64,62 @@ class _EventDetailPageConsumerState extends ConsumerState<EventDetailPage> {
   @override
   Widget build(BuildContext context) {
     final calEventLoader = ref.watch(calendarEventProvider(widget.calendarId));
-    return Scaffold(
-      body: calEventLoader.when(
-        data: (calEvent) {
-          // Update event participants list
-          updateEventParticipantsList(calEvent);
-
-          return CustomScrollView(
-            slivers: [
-              _buildEventAppBar(calEvent),
-              _buildEventBody(calEvent),
-            ],
-          );
-        },
-        error: (error, stack) {
-          _log.severe('Failed to load cal event', error, stack);
-          return ErrorPage(
-            background: const EventDetailsSkeleton(),
-            error: error,
-            stack: stack,
-            textBuilder: L10n.of(context).errorLoadingEventDueTo,
-            onRetryTap: () {
-              ref.invalidate(calendarEventProvider(widget.calendarId));
-            },
-          );
+    final errored = calEventLoader.asError;
+    if (errored != null) {
+      _log.severe(
+        'Failed to load cal event',
+        errored.error,
+        errored.stackTrace,
+      );
+      return ErrorPage(
+        background: const EventDetailsSkeleton(),
+        error: errored.error,
+        stack: errored.stackTrace,
+        textBuilder: L10n.of(context).errorLoadingEventDueTo,
+        onRetryTap: () {
+          ref.invalidate(calendarEventProvider(widget.calendarId));
         },
-        loading: () => const EventDetailsSkeleton(),
+      );
+    }
+    ref.listen(calendarEventProvider(widget.calendarId), (prev, next) async {
+      final ev = next.valueOrNull;
+      if (ev != null) {
+        final participants = asDartStringList(await ev.participants());
+        _log.info('Event Participants => $participants');
+        if (!mounted) return;
+        setState(() {
```

Comment:
Are we sure this doesn't have side-implications of throwing errors as we're mutating widget state inside ```build()```? Have it wrapped with ```WidgetInstanceBindings.addPostFrameCallback((){})``` but this could be ```listenManual``` type within ```initState(){}``` ??

Reply:
`ref.listen` is meant to be used in `build`. I can wrap the set-state of course.

Reply:
@gtalha07 I refactored this as it wasn't properly working anymore with my updates. 049fd53dcc0ae100e7094de10216138e2675c91b contains a separate provider and now all is fine again. We should add tests for these widget soon, as that would have been great to catch earlier :)